tech-debt: `test_large_template` wall-clock perf bound flakes under heavy suite load

Source: `/djust-release 0.9.0rc3` pre-flight (2026-04-28).

## Symptom

`python/tests/test_template_variable_extraction.py::TestPerformance::test_large_template` flakes when run as part of the full `make test` suite under load:

```
E   AssertionError: Extraction took 0.148s, expected < 0.1s
E   assert 0.14786410331726074 < 0.1
```

In isolation: passes in 0.06s (40% under bound). The flake is wall-clock pressure from concurrent test execution.

## Same class as #1160 (already fixed via docstring softening in v0.9.2 PR #1181)

`#1160` documented the identical pattern for `test_redis_serialization_performance`: 100ms wall-clock bound flakes under load even when isolation passes ≪10ms. The v0.9.2 fix was to soften the docstring so the test claim matches what the bound actually catches (catastrophic regressions, not perf SLA).

Apply the same pattern here:
1. Soften docstring on `test_large_template`: '100ms upper bound is a smoke test against catastrophic regression — not a perf SLA. Wall-clock 100ms can flake under heavy parallel suite load even when isolation passes in &lt;10ms.'
2. Optionally raise the bound to 200ms or 300ms for additional headroom (still catches 10x regressions).
3. OR: median-based assertion (median of N runs &lt; bound) for less wall-clock sensitivity.

Path 1 is the cheapest unblock; Path 3 is the cleaner long-term answer.

## Why not blocking v0.9.0rc3

Same posture as the v0.9.2 #1160 fix: this is wall-clock flake, not test-pollution or real perf regression. Re-run `make test` immediately after the failure → clean. The release proceeds with the flake tracked as a known issue.

## Scope

Single-file edit to `python/tests/test_template_variable_extraction.py` — docstring + optionally relaxed bound. ~5 LoC. Could bundle with #1188 in a v0.9.4 test-infra cleanup mini-drain.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -7,6 +7,30 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+
+- **v0.9.4 test-infra polish (closes #1188, closes #1189)** — three
+  small follow-ups bundled as one PR:
+  - **#1188 🟡 #1**: narrowed `vitest.config.js` Pattern 2 filter to
+    match only the diagnosed `Closing rpc` + `onUserConsoleLog` /
+    `onConsoleLog` cause from PR #1187. Dropped the broader
+    `stack.includes('view-transitions')` disjunct so future
+    genuinely-different failure shapes in `view-transitions.test.js`
+    can no longer be silently swallowed.
+  - **#1188 🟡 #2**: added `gc.collect()` before the
+    `_wait_for_one`-warning absence check in
+    `tests/integration/test_chunks_overlap.py::test_cancel_does_not_leak_wait_for_one_warning`.
+    The warning fires from CPython's coroutine GC, not explicit code;
+    the prior test passed by accident of CPython's reference-counting
+    timing. Forcing collection makes the assertion deterministic
+    under PyPy / free-threaded / different GC modes.
+  - **#1189**: bumped `test_large_template` wall-clock bound from
+    100ms → 500ms with a comment explaining the test is a regression
+    bound, not a benchmark. The prior tight bound flaked on busy CI
+    runners (5-10ms typical local; 100ms+ under py3.13 free-threaded
+    parallel suite load). Real perf tracking lives in
+    pytest-benchmark, not this assertion.
+
 ### Changed
 
 - **HVR auto-enabled in DEBUG (no AppConfig.ready() boilerplate

--- a/python/tests/test_template_variable_extraction.py
+++ b/python/tests/test_template_variable_extraction.py
@@ -355,7 +355,18 @@ class TestPerformance:
     """Test performance characteristics."""
 
     def test_large_template(self):
-        """Test extraction from large template completes quickly."""
+        """Test extraction from a 100-block template completes within a
+        generous wall-clock budget — bounding regression, not a benchmark.
+
+        The ceiling is intentionally generous (5x the typical local run)
+        because wall-clock is sensitive to parallel suite load + GC
+        scheduling under py3.13+ free-threaded mode. We don't need a
+        tight bound here; a future PR that doubles or 10×s the cost
+        would still trip this. For real perf tracking, see
+        ``pytest-benchmark`` runs in CI. Closes #1189 — the prior 100ms
+        bound flaked on busy CI runners and the v0.9.0rc3 retry surfaced
+        the same class of flake.
+        """
         import time
 
         # Generate a large template
@@ -374,8 +385,11 @@ class TestPerformance:
         result = extract_template_variables(template)
         elapsed = time.time() - start
 
-        # Should complete in less than 100ms
-        assert elapsed < 0.1, f"Extraction took {elapsed:.3f}s, expected < 0.1s"
+        # Generous regression bound: 500ms (real local runs are ~5-10ms;
+        # CI/free-threaded variance dictates the headroom). A future
+        # algorithmic regression that pushes this above 500ms is a real
+        # signal worth catching.
+        assert elapsed < 0.5, f"Extraction took {elapsed:.3f}s, expected < 0.5s"
 
         # Verify we got results
         assert len(result) > 0

--- a/python/tests/test_template_variable_extraction.py
+++ b/python/tests/test_template_variable_extraction.py
@@ -356,16 +356,18 @@ class TestPerformance:
 
     def test_large_template(self):
         """Test extraction from a 100-block template completes within a
-        generous wall-clock budget — bounding regression, not a benchmark.
+        catastrophic-regression budget — a flake-resistant smoke check,
+        not a benchmark.
 
-        The ceiling is intentionally generous (5x the typical local run)
-        because wall-clock is sensitive to parallel suite load + GC
-        scheduling under py3.13+ free-threaded mode. We don't need a
-        tight bound here; a future PR that doubles or 10×s the cost
-        would still trip this. For real perf tracking, see
-        ``pytest-benchmark`` runs in CI. Closes #1189 — the prior 100ms
-        bound flaked on busy CI runners and the v0.9.0rc3 retry surfaced
-        the same class of flake.
+        Typical local runtime is well under 1ms (~0.5ms warm). The 500ms
+        ceiling is set to absorb worst-case CI variance under busy
+        parallel suite load + py3.13 free-threaded scheduling, not to
+        catch fine-grained regressions. A future change that introduces
+        a ~1000× slowdown (algorithmic regression, accidental N²) will
+        still trip this; small constant-factor regressions will not.
+        For tight perf tracking use ``pytest-benchmark`` runs in CI.
+        Closes #1189 — the prior 100ms bound flaked on busy runners and
+        the v0.9.0rc3 retry surfaced the same class of flake.
         """
         import time
 
@@ -385,10 +387,10 @@ class TestPerformance:
         result = extract_template_variables(template)
         elapsed = time.time() - start
 
-        # Generous regression bound: 500ms (real local runs are ~5-10ms;
-        # CI/free-threaded variance dictates the headroom). A future
-        # algorithmic regression that pushes this above 500ms is a real
-        # signal worth catching.
+        # Catastrophic-regression bound: 500ms (typical local ~0.5ms;
+        # CI / free-threaded variance dictates the wide headroom).
+        # Sized to catch ~1000× regressions, not constant-factor ones —
+        # see docstring.
         assert elapsed < 0.5, f"Extraction took {elapsed:.3f}s, expected < 0.5s"
 
         # Verify we got results

--- a/tests/integration/test_chunks_overlap.py
+++ b/tests/integration/test_chunks_overlap.py
@@ -14,6 +14,7 @@ These tests are wall-clock-sensitive but use generous tolerances
 from __future__ import annotations
 
 import asyncio
+import gc
 import time
 import warnings
 from typing import List
@@ -262,6 +263,15 @@ class TestParallelRender:
             await asyncio.wait_for(render_task, timeout=0.5)
             await emitter.close()
             await consumer
+
+            # The `_wait_for_one was never awaited` warning fires from
+            # CPython's coroutine GC, not from explicit code. Without an
+            # explicit gc.collect() the test passes today by accident of
+            # CPython's reference-counting timing — under PyPy or a
+            # different GC mode the orphan coroutine may not be finalized
+            # before the assertion runs. Force collection here so the
+            # absence-assertion below is deterministic. Closes #1188 🟡 #2.
+            gc.collect()
 
         wait_for_one_warnings = [w for w in captured if "_wait_for_one" in str(w.message)]
         assert not wait_for_one_warnings, (

--- a/vitest.config.js
+++ b/vitest.config.js
@@ -49,11 +49,15 @@ export default defineConfig({
       }
 
       // Pattern 2: view-transitions teardown (#1152)
+      // Match only the diagnosed cause: vitest's `Closing rpc` teardown
+      // signature paired with the specific console-log RPC handler that
+      // had a pending call when the environment was torn down. The
+      // earlier broader `stack.includes('view-transitions')` disjunct
+      // (PR #1187) widened the net beyond the diagnosed shape — closes
+      // #1188 🟡 #1.
       if (
         msg.includes('Closing rpc') &&
-        (msg.includes('onUserConsoleLog') ||
-          msg.includes('onConsoleLog') ||
-          stack.includes('view-transitions'))
+        (msg.includes('onUserConsoleLog') || msg.includes('onConsoleLog'))
       ) {
         return false;
       }

